features/livepatch.feature

```diff
@@ -73,14 +74,46 @@ Feature: Livepatch
         Supported livepatch kernels are listed here: https://ubuntu.com/security/livepatch/docs/kernels
         """
         Examples: ubuntu release
-            | release |
-            | focal   |
+            | release | machine_type |
+            | focal   | lxd-vm       |
+
+    @series.focal
+    @uses.config.machine_type.any
+    @uses.config.machine_type.gcp.generic
+    Scenario Outline: Attached livepatch status shows upgrade required when on an old kernel
+        Given a `<release>` `<machine_type>` machine with ubuntu-advantage-tools installed
+        When I attach `contract_token_staging` with sudo
+        When I run `apt-get install linux-headers-<old_kernel_version> linux-image-<old_kernel_version> -y` with sudo
+        When I run `DEBIAN_FRONTEND=noninteractive apt-get remove linux-image*-gcp -y` with sudo
+        When I run `update-grub` with sudo
+        When I reboot the machine
+        When I run `uname -r` with sudo
+        Then stdout contains substring:
+        """
+        <old_kernel_version>
+        """
+        When I run `pro status` with sudo
+        Then stdout matches regexp:
+        """
+        livepatch +yes +warning +Canonical Livepatch service
+        """
+        Then stdout contains substring:
```

Comment:
Should we add the upgrade step just to guarantee that we are getting rid of the notice ?

Reply:
yep good idea!